core/src/main/java/org/apache/spark/SparkThrowable.java

```diff
@@ -42,6 +42,8 @@ default String getSqlState() {
     return SparkThrowableHelper.getSqlState(this.getErrorClass());
   }
 
+  default QueryContext getQueryContext() { return null; }
```

Comment:
I should add some test cases for this. Will do it later.

Reply:
shall we return `QueryContext[]` here? The error context can be stacked, e.g. view referencing other views.

Reply:
OK, it requires efforts to refactor TreeNode.origin as `Seq[Origin]` from `Origin`. But let's make it an array in the interface.